### PR TITLE
feat(annotations): Integrate annotation api with Feed

### PR DESCRIPTION
**Pull request by ConradJChan**

### CODE REVIEW

File: src/api/Feed.js [hunk 1]
```diff
@@ -227,26 +233,46 @@ class Feed extends Base {
...
+    fetchAnnotations(): Promise<?Annotation> {
+        this.annotationsAPI = new AnnotationsAPI(this.options);
+        return new Promise(resolve => {
+            this.annotationsAPI.getAnnotations(
+                this.file.id,
+                undefined,
+                resolve,
+                this.fetchFeedItemErrorCallback.bind(this, resolve),
+                1000,
+                false,
```
Comment (jstoffan) [inline src/api/Feed.js:273]: Still needed?

File: src/utils/__tests__/sorter.test.js [hunk 2]
```diff
@@ -295,11 +295,43 @@ describe('util/sorter', () => {
                 },
             ],
         };
+        const annotations = {
```
Comment (jstoffan) [inline src/utils/__tests__/sorter.test.js:298]: Do we want to move these mocks to a separate file so they can be reused? Looks like the same data is in a couple places now.

**Comment by ConradJChan** [comment]
@jstoffan it does fail gracefully, but I think I'll protect it with the feature framework for now, thanks for catching that

File: src/utils/__tests__/sorter.test.js (hunk 2 above)
Comment (ConradJChan) [reply src/utils/__tests__/sorter.test.js:299]: Good idea

File: src/__mocks__/annotations.js [hunk 3]
```diff
@@ -0,0 +1,28 @@
+const mockAnnotation = {
+    created_at: 'Thu Sep 20 33658 19:45:39 GMT-0600 (CST)',
+    creatd_by: { name: 'John', id: '987' },
+    description: 'mock annotation',
```
Comment (jstoffan) [inline src/__mocks__/annotations.js:4]: This is actually a Reply object:

```
description: {
    message: 'test message',
    type: 'reply',
}

File: src/__mocks__/annotations.js [hunk 4]
```diff
@@ -0,0 +1,28 @@
...
+    file_version: {
+        id: '123',
```
Comment (jstoffan) [inline src/__mocks__/annotations.js:6]: Missing `type: 'file_version'`, but probably not a huge deal.

File: src/__mocks__/annotations.js [hunk 5]
```diff
@@ -0,0 +1,28 @@
...
+    },
+    id: '135',
+    modified_at: 'Thu Sep 20 33658 19:46:39 GMT-0600 (CST)',
```
Comment (jstoffan) [inline src/__mocks__/annotations.js:9]: I think this is returned in a different format.

File: src/__mocks__/annotations.js [hunk 6]
```diff
@@ -0,0 +1,28 @@
...
+        },
+        shape: {
+            type: 'rect',
+            height: 10,
+            width: 10,
+            x: 1,
+            y: 1,
+        },
+        type: 'region',
+    },
```
Comment (jstoffan) [inline src/__mocks__/annotations.js:25]: Missing `type: 'annotation'`

File: src/api/__tests__/Feed.test.js [hunk 7]
```diff
@@ -98,6 +99,11 @@ const versionsWithCurrent = {
     entries: [mockCurrentVersion, mockFirstVersion, deleted_version],
 };
 
+const annotations = {
+    total_count: 1,
```
Comment (jstoffan) [inline src/api/__tests__/Feed.test.js:103]: Does this get returned by the API or is it added client-side? It's a marker-based API, so it doesn't usually know the total item count.

File: src/utils/__tests__/sorter.test.js [hunk 8]
```diff
@@ -295,11 +296,16 @@ describe('util/sorter', () => {
...
+            total_count: 1,
```
Comment (jstoffan) [inline src/utils/__tests__/sorter.test.js:300]: Stray `total_count`.

**Pull request merged**